tests/integration/targets/aci_interface_config/tasks/main.yml

```diff
@@ -0,0 +1,619 @@
+# Test code for the ACI modules
+# Copyright: (c) 2023, Sabari Jaganathan <sajagana@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("debug") }}'
+
+- name: Query system information
+  cisco.aci.aci_system:
+    <<: *aci_info
+    id: 1
+    state: query
+  register: version
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for ACI v6+ and non-cloud sites
+  when:
+  - version.current.0.topSystem.attributes.version is version('6', '>=')
+  - query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:
+    - name: Ensure Interface does not exists, invalid test without node id
+      cisco.aci.aci_interface_config:
+        <<: *aci_info
+        role: leaf
+        port_type: access
+        interface_type: switch_port
+        interface: "1/1/1"
+        state: absent
+      register: invalid_interface_absent
+      ignore_errors: yes
+
+    - name: Ensure Interface 501 does not exists, invalid test with invalid interface details
+      cisco.aci.aci_interface_config:
+        <<: *aci_info
+        role: leaf
+        port_type: access
+        interface_type: switch_port
+        node: 501
+        interface: "a/b/c"
+        state: absent
+      register: invalid_interface_details
+      ignore_errors: yes
+
+    - name: Add Interface 501, invalid test with policy_group and breakout
+      cisco.aci.aci_interface_config:
+        <<: *aci_info
+        role: leaf
+        port_type: access
+        interface_type: switch_port
+        policy_group: ans_test_switch_port
+        node: 501
+        breakout: "100g-4x"
+        interface: "1/1/1"
+        state: present
+      register: invalid_policy_group_with_breakout
+      ignore_errors: yes
+
+    - name: Assertions check for the invalid tasks
+      assert:
+        that:
+          - invalid_interface_absent.msg is match("state is absent but all of the following are missing{{':'}} node")
+          - invalid_interface_details is not changed
+          - invalid_interface_details.msg is match("MODULE FAILURE\nSee stdout/stderr for the exact error")
```

Comment:
should our module allow these type of errors? think this could be avoided with try/catch in the code